[DOC] emptyDir limitation for some plugins

**Is your feature request related to a problem? Please describe.**
Configuring Pulp operator with emptyDir will fails the execution of some plugins.
For example, `pulp-container` should access the data created by `pulpcore-api` component through `pulpcore-content` pod.
In an environment with emptyDir, each container will have its own "mount point", which will fail some plugin execution.

**Describe the solution you'd like**
We should document the above "limitation" and reiterate that emptyDir should **not** be used in production environments.

**Describe alternatives you've considered**
Not sure if we could add this in the FAQ section, but here is an example of an error that can happen because of this "wrong" config:

```
$ skopeo --debug sync --dest-creds $USER:$PWD --src docker --dest docker --dest-tls-verify=false http://quay.io/fedora/fedora-minimal:latest galaxy.apps-crc.testing/fedora
...
[2022-12-05 12:54:32 +0000] [58] [ERROR] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/local/lib/python3.8/site-packages/pulpcore/content/authentication.py", line 41, in authenticate
    return await handler(request)
  File "/usr/local/lib/python3.8/site-packages/pulp_container/app/registry.py", line 151, in get_tag
    return await self.dispatch_tag(request, tag, response_headers)
  File "/usr/local/lib/python3.8/site-packages/pulp_container/app/registry.py", line 178, in dispatch_tag
    return await Registry._dispatch(artifact, response_headers)
  File "/usr/local/lib/python3.8/site-packages/pulp_container/app/registry.py", line 78, in _dispatch
    raise Exception("Expected path '{}' is not found".format(path))
Exception: Expected path '/var/lib/pulp/media/artifact/44/3f32fc0d4897624e1ef95fd079002c5b5312ba6743a51badba9dc8c538057b' is not found
::ffff:10.217.0.1 [05/Dec/2022:12:54:32 +0000] "GET /pulp/container/fedora/fedora-minimal/manifests/latest?validate_token=99d67d536e8f97b688b6e5e3c1672c084c45b14a1c0783d5aa1b641eb36d3d30:5a5c325ea5c2e68b2a6285ef1f654f24433d8794b17c8059f02b5d8b7dbbd229 HTTP/1.1" 500 244 "https://galaxy.apps-crc.testing/v2/fedora/fedora-minimal/manifests/latest" "skopeo/1.10.0"
[2022-12-05 12:54:33 +0000] [57] [ERROR] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib64/python3.8/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/local/lib/python3.8/site-packages/pulpcore/content/authentication.py", line 41, in authenticate
    return await handler(request)
  File "/usr/local/lib/python3.8/site-packages/pulp_container/app/registry.py", line 253, in get_by_digest
    return await Registry._dispatch(artifact, headers)
  File "/usr/local/lib/python3.8/site-packages/pulp_container/app/registry.py", line 78, in _dispatch
    raise Exception("Expected path '{}' is not found".format(path))
Exception: Expected path '/var/lib/pulp/media/artifact/7a/ba5b75f1fd5f3c7f4ca331507d48e2987b567b96b40bf901b5a2a37255cb11' is not found
::ffff:10.217.0.1 [05/Dec/2022:12:54:32 +0000] "HEAD /pulp/container/fedora/fedora-minimal/blobs/sha256:7aba5b75f1fd5f3c7f4ca331507d48e2987b567b96b40bf901b5a2a37255cb11?validate_token=9849ba60bcd5089a17c0a62b7c84ffaa362f931e27accc62c1a0e97bcd848d18:1d928a18ba3ecf5bb3df33b1e5f623598ea86725606633220ba7bb67aa185ce0 HTTP/1.1" 500 189 "https://galaxy.apps-crc.testing/v2/fedora/fedora-minimal/blobs/sha256:7aba5b75f1fd5f3c7f4ca331507d48e2987b567b96b40bf901b5a2a37255cb11" "skopeo/1.10.0"
```

--- a/CHANGES/824.doc
+++ b/CHANGES/824.doc
@@ -1,0 +1,1 @@
+Added more information regarding the usage and limitation of `emptyDir`.

--- a/docs/configuring/storage.md
+++ b/docs/configuring/storage.md
@@ -38,6 +38,18 @@ If the Kubernetes cluster has no Storage Class configured, it is possible to con
 !!! note
     If the Storage Class defined will provision RWO volumes, it is recommended to also set the [`Deployment strategy`](https://docs.pulpproject.org/pulp_operator/pulp/) in Pulp CR as [`Recreate`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#recreate-deployment) to avoid the [`Multi-Attach`](https://docs.pulpproject.org/pulp_operator/faq/#how-can-i-fix-the-multi-attach-error-for-volume-my-volume-volume-is-already-used-by-pods-my-pod) volume error.
 
+Here is an example to deploy Pulp in a persistent way providing different `StorageClasses` for each component:
+```
+spec:
+  file_storage_storage_class: my-sc-for-pulpcore
+  file_storage_size: "10Gi"
+  file_storage_access_mode: "ReadWriteMany"
+  database:
+    postgres_storage_class: my-sc-for-database
+  cache:
+    redis_storage_class: my-sc-for-cache
+```
+
 
 ## Configuring Pulp Operator storage to use a Persistent Volume Claim
 
@@ -59,6 +71,16 @@ If the installation namespace has no Persistent Volume Claim available, it is po
 
 !!! note
     If the Persistent Volume Claim defined is bound to a RWO volume, it is recommended to also set the [`Deployment strategy`](https://docs.pulpproject.org/pulp_operator/pulp/) in Pulp CR as [`Recreate`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#recreate-deployment) to avoid the [`Multi-Attach`](https://docs.pulpproject.org/pulp_operator/faq/#how-can-i-fix-the-multi-attach-error-for-volume-my-volume-volume-is-already-used-by-pods-my-pod) volume error.
+
+Here is an example to deploy Pulp in a persistent way providing different `PersistentVolumeClaims` for each component:
+```
+spec:
+  pvc: my-pvc-for-pulpcore
+  database:
+    pvc: my-pvc-for-database
+  cache:
+    pvc: my-pvc-for-cache
+```
 
 ## Configuring Pulp Operator to use object storage
 
@@ -172,4 +194,10 @@ You must configure storage for the Pulp Operator. For non-production clusters, y
 
 !!! warning
     Configure this option for only non-production clusters.
+
+!!! warning
+    The content stored in an `emptyDir` volume is not shared between the pods, because of that deploying Pulp with more than a single replica of `pulpcore-api` and/or `pulpcore-content` will result in unexpected behaviors.
+
+Configuring Pulp operator with `emptyDir` will fail the execution of some plugins.
+For example, `pulp-container` plugin needs to access the data created by `pulpcore-api` component through `pulpcore-content` pod, but since each pod has its own `emptyDir` volume - and their data is not shared between them - Pulp will not work as expected.
 

